internal/auth/providers/amazon_cognito.go

```diff
@@ -0,0 +1,472 @@
+package providers
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/buzzfeed/sso/internal/auth/circuit"
+	"github.com/buzzfeed/sso/internal/pkg/groups"
+	log "github.com/buzzfeed/sso/internal/pkg/logging"
+	"github.com/buzzfeed/sso/internal/pkg/sessions"
+	"github.com/datadog/datadog-go/statsd"
+	"golang.org/x/xerrors"
+)
+
+// AmazonCognitoProvider is an implementation of the Provider interface.
+type AmazonCognitoProvider struct {
+	*ProviderData
+	StatsdClient *statsd.Client
+	AdminService CognitoAdminProvider
+	cb           *circuit.Breaker
+	GroupsCache  groups.MemberSetCache
+}
+
+type GetCognitoUserProfileResponse struct {
```

Comment:
Since we only use this struct in this pkg lets not export it by making it lowercase. `getCognitoUserProfileResponse`